Sources/SwiftDependencyGraphs/DependencyGraph.swift

```
@@ -105,6 +105,46 @@ public struct DependencyGraph<V> where V: Hashable, V: Identifiable {
     return edges[vertex.id]
   }
 
+  public typealias RemoveVertexResult = Result<RemoveVertexSuccess<V>, RemoveVertexError<V>>
+
+  @discardableResult mutating public func remove(vertex: V, byForce isForced: Bool = false) -> RemoveVertexResult {
```

`remove(vertex:)`  sounds clear -- at least as an API user I assume that it does what it says. The `byForce:` argument is interesting -- if anything, I'd like to read a sentence about its use. (IMHO no need to doc-comment the rest if that one is clear)

```
/// - Parameter isForced: ....
```

---

I added a doc-comment for the method.

---

Nice!

> Whether the vertex is removed by force. Defaults to false. If `false` the vertex is removed iff it does not have edges to other vertices. If `true` the edges to other vertices are also removed

I believe the following is a good idea, but it's also a nit; sharing it because we talked about writing challenges:  As a reader, I was surprised about what information I got first when reading this. The most interesting piece is "what happens when I enable this flag", and it comes last.

Easy reordering:

> Whether the vertex is removed by force. If `true` the edges to other vertices are also removed.  If `false` the vertex is removed iff it does not have edges to other vertices. (Defaults to false.)

From there, I see this potential:

- "Defaults to ..." was used in Apple API documentation for the longest time. Not sure if it's still needed, because for a couple of years now, we can see the default parameters in function signatures during e.g. auto-completion. I'd consider removing this part optional, it's fine to keep it. But if you tire of writing this, I believe not having these statements nowadays is not a loss anymore.
- "Whether the vertex is removed by force" restates the function and parameter names without explaining much. The explanation comes in the following sentences.
 
My attempt to rewrite:
 
> isForced: `true` to remove both `vertex` and its edges to other vertices in the graph in case `vertex` is still connected to other vertices. `false` to remove `vertex` iff  has no edges to other vertices, failing silently otherwise. (Defaults to `false`.)

Example removal functions with boolean params as inspo:

https://developer.apple.com/documentation/realitykit/entity/childcollection/removeall(keepcapacity:preservingworldtransforms:)#
https://developer.apple.com/documentation/realitykit/entity/childcollection/remove(at:preservingworldtransform:)#

---

What does _failing silently otherwise_ mean in this case? I am not sure about this because if it is not possible to remove the vertex, the function returns a `RemoveVertexError`. So, the user gets feedback.